src/Uno.UI/UI/Xaml/Controls/Image/Image.skia.cs

```diff
@@ -28,7 +28,14 @@ public Image()
 
 		partial void OnSourceChanged(ImageSource newValue, bool forceReload)
 		{
+			// We clear the old image first. We do NOT wait for the new image to load
+			// for it to replace the old one. This is what happens on WinUI.
 			_sourceDisposable.Disposable = null;
+			_lastMeasuredSize = default;
+			_imageSprite.Brush = null;
+			_currentSurface = null;
+			_pendingImageData = new();
+			InvalidateMeasure();
```

Comment:
@ramezgerges Can you try on winui to see if, when using a very slow URL, the image disapears immediately (if there already was an image displayed)? I recall some other types of behaviors around this on iOS/Android.

Reply:
Yes, WinUI will immediately remove the old image (and its allocated area) and then acquire the image area back once the new image loads. It looks pretty jarring.

Reply:
@ramezgerges I vaguely recall issues around databinding in item templates, particularly around databinding the source. Is this the scenario that you tested ?

Reply:
I checked with a binding and it's behaving the same as you described.